test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/CreatedAtRouteResultTests.cs

```diff
@@ -0,0 +1,141 @@
+
+            // Act & Assert
+            await ExceptionAssert.ThrowsAsync<InvalidOperationException>(
+                async () => await result.ExecuteResultAsync(actionContext),
+            "No route matches the supplied values.");
+        }
+
+        private static HttpResponse GetMockedHttpResponseObject()
+        {
+            var stream = new MemoryStream();
+            var httpResponse = new Mock<HttpResponse>();
+            httpResponse.SetupProperty(o => o.StatusCode);
+            httpResponse.Setup(o => o.Headers).Returns(
+                new HeaderDictionary(new Dictionary<string, string[]>()));
+            httpResponse.SetupGet(o => o.Body).Returns(stream);
+            return httpResponse.Object;
+        }
+
+        private static ActionContext GetActionContext(HttpContext httpContext)
+        {
+            var routeData = new RouteData();
+            routeData.Routers.Add(new Mock<IRouter>().Object);
+
+            return new ActionContext(httpContext,
+                                    routeData,
+                                    new ActionDescriptor());
+        }
+
+        private static HttpContext GetHttpContext(string appRoot, HttpResponse response)
+        {
+            var httpContext = new Mock<HttpContext>();
+            var actionContext = GetActionContext(httpContext.Object);
+            var mockContentAccessor = new Mock<IContextAccessor<ActionContext>>();
+            mockContentAccessor.SetupGet(o => o.Value).Returns(actionContext);
+            var mockActionSelector = new Mock<IActionSelector>();
+
+            httpContext.Setup(o => o.Response)
+                       .Returns(response);
+            httpContext.Setup(o => o.RequestServices.GetService(typeof(IOutputFormattersProvider)))
+                       .Returns(new TestOutputFormatterProvider());
+            httpContext.Setup(o => o.Request.PathBase)
+                       .Returns(new PathString(appRoot));
+
+            return httpContext.Object;
+        }
+
+        private static IUrlHelper GetMockUrlHelper(string returnValue)
+        {
+            var urlHelper = new Mock<IUrlHelper>();
+            urlHelper.Setup(o => o.RouteUrl(It.IsAny<string>(), It.IsAny<object>(), It.IsAny<string>(),
+                It.IsAny<string>(), It.IsAny<string>())).Returns(returnValue);
+
+            return urlHelper.Object;
+        }
+
+        public static IEnumerable<object[]> CreatedAtRouteData
+        {
+            get
+            {
+                yield return new object[] { null };
+                yield return
+                    new object[] {
+                        new Dictionary<string, string>() { { "hello", "world" } }
+                    };
+                yield return
+                    new object[] {
+                        new RouteValueDictionary(new Dictionary<string, string>() {
+                                                        { "test", "case" }, { "sample", "route" } })
```

Comment:
cleanup whitespace here. If you need a linebreak then put similar things on their own lines
```
new RouteValueDictionary(new Dictionary<string, string>() {
  { "test", "case" }, 
  { "sample", "route" } 
})
```